File: Documentation/5.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Maintenance/Indexes/SetLockMode.cs
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Indexes;
using Raven.Client.Documents.Operations.Indexes;

namespace Raven.Documentation.Samples.ClientApi.Operations.Maintenance.Indexes
{
    public class SetLockMode
    {
        public SetLockMode()
        {
            using (var store = new DocumentStore())
            {
                #region set_lock_single
                // Define the set lock mode operation
                // Pass index name & lock mode
                var setLockModeOp = new SetIndexesLockOperation("Orders/Totals", IndexLockMode.LockedIgnore);

                // Execute the operation by passing it to Maintenance.Send
                // An exception will be thrown if index does not exist
                store.Maintenance.Send(setLockModeOp);
                
                // Lock mode is now set to 'LockedIgnore'
                // Any modifications done now to the index will Not be applied, and will Not throw
                #endregion
            }

            using (var store = new DocumentStore())
            {
                #region set_lock_multiple
                // Define the index list and the new lock mode:
                var parameters = new SetIndexesLockOperation.Parameters {
                    IndexNames = new[] {"Orders/Totals", "Orders/ByCompany"}, 
                    Mode = IndexLockMode.LockedError
                };
                    
                // Define the set lock mode operation, pass the parameters
                var setLockModeOp = new SetIndexesLockOperation(parameters);
                
                // Execute the operation by passing it to Maintenance.Send
                // An exception will be thrown if any of the specified indexes do not exist
                store.Maintenance.Send(setLockModeOp);
                
                // Lock mode is now set to 'LockedError' on both indexes
                // Any modifications done now to either index will throw
                #endregion
            }
        }
        
        public async Task SetLockModeAsync()
        {
            using (var store = new DocumentStore())
            {
                {
                    #region set_lock_single_async
                    // Define the set lock mode operation
                    // Pass index name & lock mode
                    var setLockModeOp = new SetIndexesLockOperation("Orders/Totals", IndexLockMode.LockedIgnore);

                    // Execute the operation by passing it to Maintenance.SendAsync
                    // An exception will be thrown if index does not exist
                    await store.Maintenance.SendAsync(setLockModeOp);
                    
                    // Lock mode is now set to 'LockedIgnore'
                    // Any modifications done now to the index will Not be applied, and will Not throw
                    #endregion
                }

                {
                    #region set_lock_multiple_async
                    // Define the index list and the new lock mode:
                    var parameters = new SetIndexesLockOperation.Parameters {
                        IndexNames = new[] {"Orders/Totals", "Orders/ByCompany"}, 
                        Mode = IndexLockMode.LockedError
                    };
                    
                    // Define the set lock mode operation, pass the parameters
                    var setLockModeOp = new SetIndexesLockOperation(parameters);
                
                    // Execute the operation by passing it to Maintenance.SendAsync
                    // An exception will be thrown if any of the specified indexes do not exist
                    await store.Maintenance.SendAsync(setLockModeOp);
                    
                    // Lock mode is now set to 'LockedError' on both indexes
                    // Any modifications done now to either index will throw
                    #endregion
                }
            }
        }
        
        private interface IFoo
        {
            /*
            #region syntax_1
            // Available overloads:
            public SetIndexesLockOperation(string indexName, IndexLockMode mode);
            public SetIndexesLockOperation(Parameters parameters);
            #endregion
            */
        }

        private class Mode
        {
            #region syntax_2
            public enum IndexLockMode
            {
                Unlock,
                LockedIgnore,
                LockedError
            }
            #endregion
        }

        #region syntax_3
        public class Parameters
        {
            public string[] IndexNames { get; set; }
            public IndexLockMode Mode { get; set; }
        }
        #endregion
    }
}

File: Documentation/6.0/Samples/php/ClientApi/Operations/Maintenance/Indexes/SetLockMode.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexLockMode;
use RavenDB\Documents\Operations\Indexes\IndexLockParameters;
use RavenDB\Documents\Operations\Indexes\SetIndexesLockOperation;

class SetLockMode
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region set_lock_single
            // Define the set lock mode operation
            // Pass index name & lock mode
            $setLockModeOp = new SetIndexesLockOperation("Orders/Totals", IndexLockMode::lockedIgnore());

            // Execute the operation by passing it to Maintenance.Send
            // An exception will be thrown if index does not exist
            $store->maintenance()->send($setLockModeOp);

            // Lock mode is now set to 'LockedIgnore'
            // Any modifications done now to the index will Not be applied, and will Not throw
            # endregion
        } finally {
            $store->close();
        }

        $store = new DocumentStore();
        try {
            # region set_lock_multiple
            // Define the index list and the new lock mode:
            $parameters = new IndexLockParameters();
            $parameters->setIndexNames([ "Orders/Totals", "Orders/ByCompany" ]);
            $parameters->setMode(IndexLockMode::lockedError());

            // Define the set lock mode operation, pass the parameters
            $setLockModeOp = new SetIndexesLockOperation($parameters);

            // Execute the operation by passing it to Maintenance.Send
            // An exception will be thrown if any of the specified indexes do not exist
            $store->maintenance()->send($setLockModeOp);

            // Lock mode is now set to 'LockedError' on both indexes
            // Any modifications done now to either index will throw
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax_1
    // Available overloads:
    SetIndexesLockOperation(?string $indexName, ?IndexLockMode $mode);
    SetIndexesLockOperation(?Parameters $parameters);
    # endregion
}
*/

/*
# region syntax_2
class IndexLockMode
{
    public static function unlock(): IndexLockMode;
    public static function lockedIgnore(): IndexLockMode
    public static function lockedError(): IndexLockMode;

    public function isUnlock(): bool;
    public function isLockedIgnore(): bool;
    public function isLockedError(): bool;
}
# endregion
*/

/*
# region syntax_3
public class IndexLockParameters
{
    private ?StringArray $indexNames = null;
    private ?IndexLockMode $mode = null;

    // ... getters and setters
}
# endregion
*/
